src/components/badge/index.js

```diff
@@ -107,12 +112,43 @@ export default class Badge extends BaseComponent {
       width += borderWidth * 2;
       height += borderWidth * 2;
     }
-
     return {width, height};
   }
 
   renderLabel() {
-    const {label} = this.props;
+    const {labelLengthFormater} = this.props;
+    let {label} = this.props;
+
+    switch (labelLengthFormater) {
```

Comment:
I would move this whole logic to a separate function for readability

Reply:
Agree. Done.